### ISSUE TITLE
An issue related to model selection

### ISSUE THREAD

**Issue by Vanint** [facebookresearch/DomainBed#24]
Thanks for your work and code framework. 

When I run your code with the default hyperparameters, I find the performance of oracle selection is weaker than that of the training domain selection. 

Is there anything wrong?

 
![1381603161814_ pic_hd](https://user-images.githubusercontent.com/26360112/96533817-2396b700-12c1-11eb-8141-7f8a33a090f6.jpg)

**Comment by igul222** [comment]
Could you let us know exactly what you ran (commit hash and command strings) to produce this result?

**Comment by Vanint** [comment]
I seem to know what's wrong. For the oracle selection, the code only saves the model after finishing training. Just running one model may perform poorly (This is exactly what I did). Therefore, for oracle, you run 20 models with different hyperparameters, and select the best hyperparameter.

Why not also use the validation set to save the optimal model during the training?

**Comment by igul222** [comment]
In short, querying the test set too many times makes the results uninformative (e.g. with infinite queries you could just brute-force overfit). We wanted the number of test-set queries to be relatively small and decoupled from things like the checkpoint frequency. We go into more detail in the paper.

**Comment by Vanint** [comment]
Then why do you post the best performance instead of the final performance? Essentially, I think the two are the same, although I agree with the situation of brute-force overfit.

**Comment by igul222** [comment]
I'm not sure what you mean. We collect final-timestep accuracies of all models and report the max over these.

**Comment by Vanint** [comment]
I mean in the IID Accuracy Selection

**Comment by igul222** [comment]
Oh, in that case we don't use the test set at all to select checkpoints; we use a validation set split off from the training data. This means it's fair game to query the validation set as much as we want.

**Comment by Vanint** [comment]
Okay. Thanks for your quick answer.
